targets/csharp/source/source/Uunit/tests/PlayFabApiTest.cs

```diff
@@ -537,10 +480,169 @@ private void GetObjects2Continued(PlayFabResult<GetObjectsResponse> result, UUni
+            testContext.NotNull(loginTask2.Result.AuthenticationContext, "AuthenticationContext is not set for loginRequest2");
+
+            if (string.Equals(loginTask.Result.AuthenticationContext.ClientSessionTicket, loginTask2.Result.AuthenticationContext.ClientSessionTicket))
+            {
+                testContext.Fail("Multiple Login Failed AuthenticationContexts are same");
+            }
+            testContext.EndTest(UUnitFinishState.PASSED, null);
+        }
+
+        /// <summary>
+        /// CLIENT API
+        /// Test that Multiple player can do api calls
+        /// </summary>
+        [UUnitTest]
+        public void MultiplePlayerApiCall(UUnitTestContext testContext)
+        {
+            // If the setup failed to log in a user, we need to create one.
+            var loginRequest = new LoginWithCustomIDRequest
+            {
+                TitleId = PlayFabSettings.TitleId,
+                CustomId = PlayFabSettings.BuildIdentifier,
+                CreateAccount = true,
+            };
+
+            var loginRequest2 = new LoginWithCustomIDRequest
+            {
+                TitleId = PlayFabSettings.TitleId,
+                CustomId = PlayFabSettings.BuildIdentifier,
+                CreateAccount = true,
+            };
+
+            var loginTask = PlayFabClientAPI.LoginWithCustomIDAsync(loginRequest, null, testTitleData.extraHeaders).Result;
+            var loginTask2 = PlayFabClientAPI.LoginWithCustomIDAsync(loginRequest2, null, testTitleData.extraHeaders).Result;
+
+            testContext.NotNull(loginTask.Result, "Login Result is null for loginRequest");
+            testContext.NotNull(loginTask2.Result, "Login Result is null for loginRequest2");
+            testContext.IsNull(loginTask.Error, "Login error occured for loginRequest: " + loginTask.Error?.ErrorMessage ?? string.Empty);
+            testContext.IsNull(loginTask2.Error, "Login error occured for loginRequest2: " + loginTask2.Error?.ErrorMessage ?? string.Empty);
+            testContext.NotNull(loginTask.Result.AuthenticationContext, "AuthenticationContext is not set for loginRequest");
+            testContext.NotNull(loginTask2.Result.AuthenticationContext, "AuthenticationContext is not set for loginRequest2");
+
+            if (string.Equals(loginTask.Result.AuthenticationContext.ClientSessionTicket, loginTask2.Result.AuthenticationContext.ClientSessionTicket))
+            {
+                testContext.Fail("MultipleApiCall Failed AuthenticationContexts are same");
+            }
+            var getPlayerProfileRequest = new GetPlayerProfileRequest()
+            {
+                AuthenticationContext = loginTask.Result.AuthenticationContext,
+                PlayFabId = PlayFabId
+            };
+            var getPlayerProfileRequest2 = new GetPlayerProfileRequest()
+            {
+                AuthenticationContext = loginTask2.Result.AuthenticationContext,
+                PlayFabId = PlayFabId
+            };
```

Comment:
Reusing AuthenticationContext from the previous test, would remove this many lines of copy-paste, which I think is a greater sin than imperfect test isolation.
As long as the first step is verifying that the context from the previous test was set correctly, it's fine.